plugin/pkg/scheduler/algorithm/scheduler_interface.go

```diff
@@ -18,10 +18,24 @@ package algorithm
 
 import (
 	"k8s.io/kubernetes/pkg/api"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
 )
 
+// SchedulerExtender is an abstract interface for external processes to influence scheduling
+// decisions made by Kubernetes.
+type SchedulerExtender interface {
+	// Filter based on extender-implemented predicate functions. The filtered list is
+	// expected to be a subset of the supplied list.
+	Filter(pod *api.Pod, nodes *api.NodeList) (*api.NodeList, error)
+
+	// Prioritize based on extender-implemented priority functions. The returned scores & weight
+	// are used to compute the weighted score for an extender. The weighted scores are added to
+	// the scores computed  by Kubernetes scheduler. The total scores are used to do the host selection.
+	Prioritize(pod *api.Pod, nodes *api.NodeList) (*schedulerapi.HostPriorityList, int, error)
+}
+
 // Scheduler is an interface implemented by things that know how to schedule pods
 // onto machines.
 type ScheduleAlgorithm interface {
-	Schedule(*api.Pod, NodeLister) (selectedMachine string, err error)
+	Schedule(*api.Pod, NodeLister, []SchedulerExtender) (selectedMachine string, err error)
```

Comment:
Hi @davidopp . Thanks for explaining it. I'm still confused that for #11793 and #17197, if multi-schedulers is supported, wouldn't this interface add more burden to other schedulers?

I think filtering and prioritization are still pretty generic. So this is still good in general.